### PR TITLE
Remove unused `Mutex_m` in Active Model

### PR DESCRIPTION
**Pull request by kamipo**

### CODE REVIEW

File: activemodel/lib/active_model/attribute_methods.rb [hunk 1]
```diff
@@ -327,13 +326,11 @@ def undefine_attribute_methods
...
       end
 
-      def generated_attribute_methods #:nodoc:
-        @generated_attribute_methods ||= Module.new {
-          extend Mutex_m
-        }.tap { |mod| include mod }
-      end
-
       private
+        def generated_attribute_methods
```
Comment (esparta) [inline activemodel/lib/active_model/attribute_methods.rb]: missing `#:nodoc:` ?

Comment (esparta) [inline activemodel/lib/active_model/attribute_methods.rb]: duhh... sorry, just realize is now private.

**Comment by rafaelfranca** [comment]
Why it is unused? It was added in https://github.com/rails/rails/commit/09b31f08d4a4930dc66d401c94b2e5e49c250fc1 and it seems it is there to a reason.

**Comment by kamipo** [comment]
`generated_attribute_methods.synchronize` is only used in `ActiveRecord::AttributeMethods` and Active Record initialize `@generated_attribute_methods` (extended by `Mutex_m`) by itself.

```
% git grep -n generated_attribute_methods.synchronize
activerecord/lib/active_record/attribute_methods.rb:56:        generated_attribute_methods.synchronize do
activerecord/lib/active_record/attribute_methods.rb:65:        generated_attribute_methods.synchronize do
```

So I think `Mutex_m` in Active Model is never used since it was added.

**Comment by rafaelfranca** [comment]
Thanks! Make sense

**Pull request merged**